### PR TITLE
Improve performance benchmark documentation

### LINKED ISSUE #3136: Document performance testing guidelines
Most of the time we use vtebench to validate the performance of Alacritty, however that's not necessarily straight-forward and there are a few pitfalls that can invalidate the results.

Over time, a relatively reliable process has been established, however it is not properly documented anywhere. We should probably document it in the Alacritty or vtebench repository and guide people on proper benchmarking procedure in our `CONTRIBUTING.md`.

### PR DESCRIPTION
**Pull request by chrisduerr**
Fixes #3136.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,9 +67,15 @@ complete correctly with the unpatched version, to make sure the test case is cov
...
+Any change which could have an impact on Alacritty's performance, should be tested to prevent
+potential regressions.
+
 Alacritty mainly uses the [vtebench](https://github.com/jwilm/vtebench) tool for testing Alacritty's
-performance. Any change which could have an impact on Alacritty's performance, should be tested with
-it to prevent potential regressions.
+performance. Instructions on how to use it can be found in its
+[README](https://github.com/jwilm/vtebench/blob/master/README.md).
+
+Latency is another important factor for Alacritty. On X11, Windows, and BSD the
```
Comment (kchibisov) [inline CONTRIBUTING.md]: s/BSD/macOS

Comment (chrisduerr) [reply CONTRIBUTING.md]: Basically the same thing, right? :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,9 +67,15 @@ complete correctly with the unpatched version, to make sure the test case is cov
...
 
+Any change which could have an impact on Alacritty's performance, should be tested to prevent
+potential regressions.
+
 Alacritty mainly uses the [vtebench](https://github.com/jwilm/vtebench) tool for testing Alacritty's
-performance. Any change which could have an impact on Alacritty's performance, should be tested with
-it to prevent potential regressions.
+performance. Instructions on how to use it can be found in its
+[README](https://github.com/jwilm/vtebench/blob/master/README.md).
+
```
Comment (kchibisov) [inline CONTRIBUTING.md]: Could we add a point to LLVM benchmarking on how to setup systems?

Comment (kchibisov) [reply CONTRIBUTING.md]: Well, having this info on vtebench side probably makes more sense... Let's just skip it for now, unless folks will be constantly ignoring this part of disabling CPU frequency scaling things.

Comment (chrisduerr) [reply CONTRIBUTING.md]: Yeah, we can only hope that people read documentation anyways, there's no guarantee no matter where we put it.

And they need to read the vtebench readme anyways, to figure out how to use it.

**Pull request merged**